### PR TITLE
refactor(fx-2731): filter q/a - autosuggest ui

### PR DESCRIPTION
**Pull request by dzucconi**
Closes: [FX-2731](https://artsyproduct.atlassian.net/browse/FX-2731)

https://github.com/artsy/force/commit/91b479c4d8b81361497ae739569c93f4c44d7e23 builds a much simpler autosuggest input. 

Since the code for all the filters is identical; I just abstracted it into a component. Mind you it's not a *good* abstraction; it's just a: this code is all the same so here it is in one place. I'm not opposed to these just being copy-paste; hence the draft. Will follow up with specs, depending on which way this goes.

### CODE REVIEW

**Review by mzikherman — COMMENTED** [review]
So, I'm generally in favor or doing something in-house where it's very simple and reasonable, such as a lil' auto-complete like this, vs using the `react-autosuggest` library.

However, in this case, I was definitely prompted to use the library due to using it in the main search bar on the site, and also the fact that you get some nice behaviors and accessibility in place for free (for instance, pretty seamless keyboard scrolling through options, hitting 'escape' to close, etc.).

It's possible in the future we'd want to back the filter autocomplete by an ES (or other external) endpoint, which....I guess this solution or keeping/refining `react-autosuggest` would both entail _some_ work, but I dunno- at the very least this PR removes the keyboard scrolling, are we cool with that?

I would sort of vote to keep using the lib, it seems like we can customize the UI a bit to match the spec w/o the rewrite. What's your take @dzucconi ?

**Comment by dzucconi** [comment]
@mzikherman in this case we actually *don't* want the "accessibility" features (which aren't even very complex in that case). We don't want arrow keyboard nav for this. You'd simply tab. What you're doing is filtering an existing UI which is the checkbox filters — It's not an auto-suggest dropdown which is a very different kind of UI. One accessibility feature I would add here is just announcing the matched results to screen readers.

**Comment by mzikherman** [comment]
> We don't want arrow keyboard nav for this.

How come? That seems like a pretty standard autocomplete/dropdown type UI interaction, which this pretty much is, no?

> It's not an auto-suggest dropdown

Ah, interesting! I'm not sure I follow, is this not that? Seems that way to me.

**Comment by dzucconi** [comment]
> It's not an auto-suggest dropdown

It's not in the sense that the non-filtered checkboxes also aren't. If one has an up/down keyboard navigation; should the other? I'm not necessarily opposed [to adding it to everything](https://github.com/dzucconi/use-keyboard-list-navigation) but it's definitely weird. There's no WAI-ARIA spec for this type of UI AFAIK. I imagine you'd want to move the focus ring around.

Could you find some prior art? Some site that does this: filters an existing checkbox UI but supports up/down keyboard navigation?

**Comment by mzikherman** [comment]
I hear you! I definitely do _not_ know any prior art for this 😅 

I'm not opposed to making the change, just wanted to understand the reasoning for stopping to use that library and everything.

**Comment by dzucconi** [comment]
I'm going to go ahead with this direction unless we want to add the up/down keyboard navigation (in which case we add but we still wouldn't want to use that library)

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ResultsFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  expanded = true,
+}) => {
+  const { aggregations, currentlySelectedFilters } = useArtworkFilterContext()
+  const { isFiltered, handleFilterChange } = useFacetFilter()
+
+  const selectedValues = currentlySelectedFilters()[facetName]
+  const results = aggregations.find(aggregation => aggregation.slice === slice)
+    ?.counts
+
+  if (!results?.length) {
```
Comment (dzucconi) [inline src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ResultsFilter.tsx:35]: Handles most of https://github.com/artsy/force/pull/7270 (sans time period filter)

**Comment by dzucconi** [comment]
@mzikherman if you wanna give this another look it should be good to go. We can always follow up with up/down arrow key nav if need be. But this is accessible as it is for the type of UI it is.

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/FacetFilter.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    setQuery(value)
+
+    onChange && onChange(event)
+
+    if (value === "") {
+      setFitleredResults([])
+      return
+    }
+
+    setFitleredResults(() =>
```
Comment (mzikherman) [inline src/v2/Components/v2/ArtworkFilter/ArtworkFilters/FacetFilter.tsx:38]: fitler

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/FacetFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      )
+    )
+  }
+
+  return (
+    <>
+      <FilterInput onChange={handleChange} mb={1} {...rest} />
+
+      {query !== "" && (
+        <VisuallyHidden aria-live="polite">
```
[the remark this answers is not in the document]
Comment (mzikherman) [reply src/v2/Components/v2/ArtworkFilter/ArtworkFilters/FacetFilter.tsx:50]: that's pretty polite

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ResultOption.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  value,
+}) => {
+  const { currentlySelectedFilters, setFilter } = useArtworkFilterContext()
+  const results = currentlySelectedFilters()[facetName]
+
+  const handleSelect = (value: string) => (selected: boolean) => {
+    setFilter(
+      facetName,
+      selected
+        ? // Append
```
Comment (mzikherman) [inline src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ResultOption.tsx:31]: nice

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/__tests__/ResultsFilter.jest.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  it("renders correctly", () => {
+    const wrapper = getWrapper({
+      aggregations: [
+        {
+          slice: "LOCATION_CITY",
+          counts: [
+            {
+              name: "Cattown, Cat City, Nowhere USA",
```
Comment (mzikherman) [inline src/v2/Components/v2/ArtworkFilter/ArtworkFilters/__tests__/ResultsFilter.jest.tsx:27]: i wonder what the real estate market here is like

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/FacetFilter.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    onChange && onChange(event)
+
+    if (value === "") {
+      setFitleredResults([])
+      return
+    }
+
+    setFitleredResults(() =>
+      results.filter(({ name }) =>
+        name.toLowerCase().includes(value.toLowerCase())
```
Comment (mzikherman) [inline src/v2/Components/v2/ArtworkFilter/ArtworkFilters/FacetFilter.tsx:40]: this is kind of an aggressive matcher, maybe `startsWith`?

**Review by mzikherman — COMMENTED** [review]
LGTM. left a couple of comments, up to you! We should prob re-QA these with a couple of peeps once it lands.

File: src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ResultsFilter.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    return null
+  }
+
+  const resultsSorted = sortBy(results, ["count"]).reverse()
+
+  const isBelowTheFoldFilterSelected =
+    intersection(
+      selectedValues,
+      results.slice(INITIAL_ITEMS_TO_SHOW).map(({ value }) => value)
```
[the remark this answers is not in the document]
Comment (dzucconi) [reply src/v2/Components/v2/ArtworkFilter/ArtworkFilters/ResultsFilter.tsx]: Woops got it thank you!

**Pull request merged**